src/keybindings.c

```diff
@@ -1827,11 +1827,11 @@ static void goto_matching_brace(GeanyDocument *doc)
 
 static gboolean cb_func_clipboard_action(guint key_id)
 {
-	GeanyDocument *doc = document_get_current();
+	GtkWidget *focusw = gtk_window_get_focus(GTK_WINDOW(main_widgets.window));
 
-	if (doc == NULL)
+	if (!IS_SCINTILLA(focusw)) {
```

Comment:
@b4n Ah, I see what you're saying now. It should support all the options which the relevant callbacks support: 
e.g.
```
	if (GTK_IS_EDITABLE(focusw))
		gtk_editable_cut_clipboard(GTK_EDITABLE(focusw));
	else if (IS_SCINTILLA(focusw))
		sci_cut(SCINTILLA(focusw));
	else if (GTK_IS_TEXT_VIEW(focusw))
```
In which case I think it might make sense to just remove the check altogether and let the individual callbacks determine what is and isn't a valid target, since they do that anyways.